benchmark_test/regression.go

```diff
@@ -0,0 +1,56 @@
+package benchmark
+
+import (
+	"fmt"
+
+	"github.com/elastic/hey-apm/errs"
+	"github.com/elastic/hey-apm/es"
+	"github.com/elastic/hey-apm/models"
+
+	"github.com/elastic/hey-apm/conv"
+	"github.com/elastic/hey-apm/types"
+	"github.com/pkg/errors"
+)
+
+// compares the given report with saved reports with the same input indexed in the last specified days
+// returns an error if connection can't be established,
+// or performance decreased by a margin larger than specified
+func VerifyNoRegression(conn es.Connection, r models.Report, margin float64, days string) error {
+	inputMap := conv.ToMap(r.Input)
+	filters := []types.M{
+		{
+			"range": types.M{
+				"@timestamp": types.M{
+					"gte": fmt.Sprintf("now-%sd/d", days),
+					"lt":  "now",
+				},
+			},
+		},
+	}
+	for k, v := range inputMap {
+		filters = append(filters, types.M{"match": types.M{k: v}})
```

Comment:
Does this mean that if an `APMServerUrl`, `APMServerSecret` etc. changes the results will no longer be comparable?

Reply:
`ApmServerUrl` yes (why wouldn't?, it would be exercising a different machine), `ApmServerSecret` no

Reply:
I missed that you only decode specific attributes of the struct, all good then 👍